app/src/ui/lib/list/selection.ts

```diff
@@ -140,3 +140,25 @@ export function findNextSelectableRow(
 
   return null
 }
+
+/**
+ * Find the last selectable row in either direction, used
+ * for moving to the first or last selectable row in a list,
+ * i.e. Home/End key navigation.
+ */
+export function findLastSelectableRow(
+  direction: SelectionDirection,
+  rowCount: number,
+  canSelectRow: (row: number) => boolean
+) {
```

Comment:
This could have been implemented also as something like to take advantage of the other method:

```ts
let row = direction === 'up' ? rowCount - 1 : 0
const traverseDirection = direction === 'up' ? 'down' : 'up'

return findNextSelectableRow(rowCount, {direction: traverseDirection, row, wrap: true }, canSelectRow)
```

Bun I honestly don't know if it's easier to read/understand LOL, so feel free to ignore this 😅

Reply:
Ah, clever you! I don't have strong feelings about it but I don't want to bother you with another review so I'm gonna merge as is 😛